services/UserAuthService.ts

```diff
@@ -42,6 +42,19 @@ export default class UserAuthService {
     });
   }
 
+  public async modifyEmail(user: UserModel, proposedEmail: string): Promise<void> {
+    return this.transactions.readWrite(async (txn) => {
+      const userRepository = Repositories.user(txn);
+
+      await userRepository.upsertUser(user, {
+        email: proposedEmail,
+        state: UserState.PENDING,
+      });
+
+      await this.setAccessCode(proposedEmail);
+    });
+  }
```

Comment:
Let's return the updated `user` object in this method instead of return void. We need the updated `email` and `accessCode` fields in `sendEmailVerification`, so it's important that we're updating the `user` param accordingly. I think the line 49 call will update `user.email`, but the line 54 call wouldn't update the  `user` param's `accessCode` field since it's not being directly set (see `setAccessCode` - it's fetching a separate object and updating that one vs. directly updating the `user` param object passed in)